crates/fs/public/file.rs

```diff
@@ -78,13 +77,9 @@ impl PublicFile {
     ///     file.store(&mut store).await.unwrap();
     /// }
     /// ```
+    #[inline(always)]
     pub async fn store<B: BlockStore>(&self, store: &mut B) -> Result<Cid> {
-        let bytes = {
-            let mut tmp = vec![];
-            self.encode(DagCborCodec, &mut tmp)?;
-            tmp
-        };
-        store.put_block(bytes, IpldCodec::DagCbor).await
+        store.put_serializable(self).await
```

Comment:
Now that this has become so much simpler, I'd vote for just having `store.put_serializable(file)` instead of both that and `file.store(store)`. I think it's important to keep the choices small, so it's clear what one should use in each situation.

Reply:
`store` is exposed as a public API. `Blockstore::put_serializable` isn't, so we need to have it.

Reply:
The issue there is we cannot expose a generic interface from WebAssembly.